### ISSUE TITLE
Non-Issue issues

### ISSUE THREAD

**Issue by royfrancis** [hartleys/QoRTs#17]
This is an excellent tool that provides pretty much all the rna-seq metrics that one would need. I have suggestions for a few things that could be improved.

+ Firstly names of function names in R package. They are lower-case (`build.plotter.basic`), camel-case (`makeMultiPlot`), mixed (`makePlot.splice.junction.event`) etc. There is probably some logic that I don't understand but I think it would be nice to just keep it uniform and simple. For example, all lower case with underscores (`make_multi_plot()`) as in the convention with the new [hadleyverse](http://adolfoalvarez.cl/the-hitchhikers-guide-to-the-hadleyverse/) packages. This would break some of my existing code, but I would be happy to change it for consistency.

+ I think this tool needs some metric for duplication rate. Perhaps this sort of plot from [dupRadar](https://www.bioconductor.org/packages/release/bioc/vignettes/dupRadar/inst/doc/dupRadar.html) or [SeqMonk](http://www.bioinformatics.babraham.ac.uk/projects/seqmonk/screenshots/duplication_plot.png).

![download](https://cloud.githubusercontent.com/assets/5083500/16039639/4c3015a6-322b-11e6-806b-c127fc562f77.png)

+ Change the rainbow palette to something more pleasant
Perhaps a colour scheme from ColorBrewer or [Viridis](https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html), `rich.colors()` from package `gplots`, `tim.colors` from package `fields` etc.

+ Summary Table
I would expect the read pairs count in summary table to add up with my idea of gene-hierarchy shown below. But, I don't think it adds up well. Then again, this could be hard with reads overlapping between features.

![gene-hierarchy](https://cloud.githubusercontent.com/assets/5083500/16040812/47c58b5c-3232-11e6-900b-03af32fec70e.png)

+ The field names in the summary table are also a mess and could be made more consistent. Perhaps some sort of regular expression so rows are easy to pull out.

+ Multiplot plot makes this big 'Other' for some reason.

![1](https://cloud.githubusercontent.com/assets/5083500/16039859/9c3f7720-322c-11e6-98e2-e911273124f0.png)

As in the title, these are minor issues but I just thought I would mention it. If I think of more, I will add them here. Thanks again for this great package.

**Comment by hartleys** [comment]
Taking the points in order:

* Function names: Changing the function names would break user code. Good programming practices dictate that I not make incompatible API-level changes without a **very** good reason. [It would also require iterating the major version number](http://semver.org/). So unfortunately at this stage we're pretty much stuck with the function names as-is, at least until the release of QoRTs2 (hypothetically).

* Duplicates: Unlike in DNA-Seq, in RNA-Seq detecting duplicates is kind of tricky. Often duplicate detection algorithms can be "tricked" by the presence of small, high-coverage transcripts. This can cause tools to falsely indicate a higher dup rate, when actually what is happening is a higher expression of one or two small transcripts. Also: the plots you show only show a single sample. We would need a way to readably overlay numerous samples on top of one another. I'm not saying no, I'm just saying it's much trickier than it appears. I welcome your input if you have ideas on how this could be done well.

* Colors: I tried colorbrewer and found that most of the color schemes were too limited. I'll take a look at the other packages you mention. I've been looking for a good automated color selection tool for a long time, so thanks for the heads up.

* I'm not sure what you mean. Why doesn't it "add up well"? Can you show me an example? The diagram looks exactly like the way I designed things.

* Field codes: Changing the field codes is also an API-level change and is not to be made lightly. Again, it would break user pipelines.

* The "big other" is due to an overzealous auto-resizer, which is resizing the word "other" to fit in the horizontal space provided. I will fix this in the next release.

**Comment by royfrancis** [comment]
For fourth point, I mean literally adding up. 

For example, here is one sample and some fields...

```
                              field  sample1
38            input.read.pair.count 77560044
61              ReadPairs_AmbigGene   174926
62                 ReadPairs_NoGene 31742205
63          ReadPairs_NoGene_Intron 18897768
64 ReadPairs_NoGene_MiddleOfNowhere  6652716
65   ReadPairs_NoGene_OneKbFromGene  1431738
66   ReadPairs_NoGene_TenKbFromGene  4759983
67             ReadPairs_UniqueGene 27866572
68         ReadPairs_UniqueGene_CDS 13829839
69         ReadPairs_UniqueGene_UTR 14036733
```

ReadPairs_NoGene+ReadPairs_UniqueGene+ReadPairs_AmbigGene != input.read.pair.count
ReadPairs_NoGene+ReadPairs_UniqueGene+ReadPairs_AmbigGene+ReadPairs_NoGene_Intron != input.read.pair.count

This one has been resolved.
ReadPairs_NoGene_TenKbFromGene+ReadPairs_NoGene_OneKbFromGene+ReadPairs_NoGene_MiddleOfNowhere != ReadPairs_NoGene
The one above was resolved because I realised that intron was considered as non-gene, thus
ReadPairs_NoGene_TenKbFromGene+ReadPairs_NoGene_OneKbFromGene+ReadPairs_NoGene_MiddleOfNowhere+ReadPairs_NoGene_Intron = ReadPairs_NoGene

**Comment by hartleys** [comment]
input.read.pair.count indicates the original read pair count, including all the unmapped and filtered-out reads. This optional field is provided by the decoder file, and is used to calculate the mapping rate stats. 

What you want is the "READ_PAIR_OK" field:

ReadPairs_NoGene+ReadPairs_UniqueGene+ReadPairs_AmbigGene == READ_PAIR_OK

By the way, you can set the colors manually by setting the plotter.params parameter. For example:

    data("res",package="QoRTsExampleData")
    makeMultiPlot.colorBySample(res, plotter.params = list(contrasting.colors = viridis(6)))

**Comment by royfrancis** [comment]
Thanks a lot :+1:

**Comment by hartleys** [comment]
The oversized x-axis label issue has been fixed in the newest release (v1.1.6).
